Kinvey.Core/Store/MultiInsertRequest.cs

```diff
@@ -125,13 +125,24 @@ public override async Task<KinveyMultiInsertResponse<T>> ExecuteAsync()
                     {
                         exception = httpRequestException;
                     }
+                    catch (KinveyException kinveyException)
+                    {
+                        exception = kinveyException;
+                    }
 
                     if (exception != null)
                     {
                         foreach (var pendingAction in pendingWriteActions)
                         {
                             SyncQueue.Enqueue(pendingAction);
-                        }                       
+                        }
+
+                        var kinveyException = exception as KinveyException;
```

Comment:
@eugeninnowise What is the reason for casting the `KinveyException` to `Exception` and then back to `KinveyException`?

Reply:
@vinaygahlawat 
The type Exception has the “exception” variable. We are casting all caught exceptions(HttpRequestException and KinveyException) to the “exception” variable.   If  exception != null then we go to the block where we are adding pending actions. In this block we are casting the "exception" back to the KinveyException to know whether  to throw the KinveyException or not.
If you don't like casting, we could have two variables with types HttpRequestException and KinveyException. It would help us avoid type casting.
Do you have other ideas?

Reply:
You could do it with two types as you said, or keep it the way it is. If you keep it the way it currently is, add a comment in the code that explains what you described above regarding (HttpRequestException and KinveyException).

Reply:
Ok. I will use the way with two types.
Thank you @vinaygahlawat  for the recommendation.